File: 프로그래머스/lv2/42587. 프로세스/프로세스.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<Pair> q = new LinkedList<Pair>(); 
        for(int i = 0; i < priorities.length; i++){
            q.add(new Pair(priorities[i],i));
        }
        while(!q.isEmpty()){
            boolean check = true;
            Pair p = q.peek();
            for(Pair pair : q){
                int val = p.value;
                if(pair.value > val){
                    check = false;
                    q.add(q.poll());
                    break;
                }
            }
            if(check){
                Pair cp = q.poll();
                answer++;
                if(location == cp.number){
                    break;
                }
            }
        }
        return answer;
    }
}
class Pair {
        int value;
        int number;

        public Pair(int value, int number) {
            this.value = value;
            this.number = number;
        }
    }

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    move = deque()
    answer = 0
    for i in range(0,len(priorities)):
        new_pair = Pair(priorities[i],i)
        move.append(new_pair)

    while move:
        check = True
        pair = move[0]
        
        for j in move:
            if j.value > pair.value:
                check = False
                move.append(move.popleft())
                break
        
        if(check):
            ck_pair = move.popleft()
            answer += 1
            if location == ck_pair.location:
                break
                
        

    return answer

class Pair:
    def __init__(self, value, location):
        self.value = value
        self.location = location
        